### PR TITLE
datadog notebooks, monitors and customizable domain

### PR DESCRIPTION
**Pull request by aexvir**
## Description

This merge request adds two new commands, allowing to list [notebooks](https://docs.datadoghq.com/notebooks/) and [monitors](https://docs.datadoghq.com/monitors/) along with their status; as well as allowing custom domains for generating the link to view resources in the browser, which is part of some enterprise plans.

The icons I've used to represent the state of the monitors are from [fontawesome](https://github.com/FortAwesome/Font-Awesome).

## Type of change

- Bug fix / improvement for somebody else's extension. Extension author: @kudrykv

## Screencast

Screenshot 2022-02-11 at 00 33 01

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and [tested this distribution build in Raycast](https://developers.raycast.com/basics/prepare-an-extension-for-store#metadata-and-configuration)

### CODE REVIEW

**Comment by pernielsentikaer** [comment]
Awesome @aexvir 🔥 

 Could you test it out @kudrykv and give some feedback?

**Comment by kudrykv** [comment]
## Monitors
Monitors are weird — some tags work, some don't. I wonder whether this is an issue of Datadog itself. Here, it shows results for the given tag:
CleanShot 2022-02-11 at 15 22 21@2x

And here it doesn't:
CleanShot 2022-02-11 at 15 23 03@2x

## Notebooks
Notebooks don't work for me for some reason:
CleanShot 2022-02-11 at 15 30 59@2x
In console I get tons of `could not deserialize` errors.
<details>
<summary>Errors snippet</summary>

```
...
2022-02-11 15:17:08.305 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize NotebookRelativeTime (Error: missing required property 'liveSpan')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize NotebookRelativeTime (Error: missing required property 'liveSpan')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize NotebookRelativeTime (Error: missing required property 'liveSpan')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize NotebookMarkdownCellAttributes (Error: missing required property 'text')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize FormulaAndFunctionApmDependencyStatsQueryDefinition (Error: missing required property 'env')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize FormulaAndFunctionApmResourceStatsQueryDefinition (Error: missing required property 'env')
2022-02-11 15:17:08.306 Db Raycast[22145:d30b31] [com.raycast.extensions.raycast-datadog:command-list-notebooks] could not deserialize FormulaAndFunctionEventQueryDefinition (Error: missing required property 'compute')
...
```

</details>

Missing required property 'id' error:
```
Error: missing required property 'id'
    at Function.ObjectSerializer2.deserialize (/Users/bgnfu7re/develop/projects/raycast-extensions/extensions/datadog/node_modules/@datadog/datadog-api-client/packages/datadog-api-client-v1/models/ObjectSerializer.ts:1753:17)
    at Function.ObjectSerializer2.deserialize (/Users/bgnfu7re/develop/projects/raycast-extensions/extensions/datadog/node_modules/@datadog/datadog-api-client/packages/datadog-api-client-v1/models/ObjectSerializer.ts:1692:28)
```

**Comment by aexvir** [comment]
hmm... I wonder if it has something to do with the scope of the tokens... both features work fine for me 🤔 
btu yeah, good point, I didn't really add any graceful handling to these features, I was just relying on the datadog client's error handling 😅 

will take a look and fix this a bit later
thanks for checking!

**Comment by kudrykv** [comment]
@aexvir, thank you!
I've just realized another point. It would be great to cache the notebook response for a while, like 1-5 minutes or so. They don't change frequently, and it would be nice to have them in an instant. After the cache expires, you can still show it until a new fetch happens, and refresh it afterward.

**Comment by aexvir** [comment]
funny enough I was thinking about exactly that as my next contribution, but for dashboards 😄️ 
no sure what's the usage pattern for notebooks, but I personally use them *a lot* so they do change quite often on my case
but yeah, a couple minutes latency shouldn't be that bad (maybe configurable?) I need to check how caching works on raycast either way

but I'd definitely keep that outside of this pr if you don't mind, let's keep this one small

**Comment by aexvir** [comment]
small screenshot for the notebooks as well, just to show how does it look
I'd like to put more screenshots but it's annoying to have to blur so much data 😅 

Screenshot 2022-02-11 at 19 34 10

**Comment by pernielsentikaer** [comment]
Awesome @aexvir 💪 

Sorry to bother you all the time @kudrykv – could you take a new look?

**Comment by kudrykv** [comment]
No problem checking at all 🙂
Yesterday was a busy day. Going to check it sometime today

**Comment by kudrykv** [comment]
> no sure what's the usage pattern for notebooks, but I personally use them a lot so they do change quite often on my case

Do you create notebooks often?

> funny enough I was thinking about exactly that as my next contribution, but for dashboards 😄️ those are really static on our case

Yeah, that makes sense, too!

**Comment by aexvir** [comment]
> Do you create notebooks often?
yep! 😅 maybe I'm overusing them, but I make a notebook pretty much every time I need to check a metric, as I find it much better than the metrics page

@pernielsentikaer anything else missing in order to get this merged?

File: extensions/datadog/package.json [hunk 1]
```diff
@@ -17,6 +18,29 @@
       "description": "List Dashboards",
       "mode": "view"
     },
+    {
+      "name": "command-list-notebooks",
+      "title": "List notebooks",
```
Comment (pernielsentikaer) [inline extensions/datadog/package.json:23]:
```suggestion
      "title": "List Notebooks",
```

File: extensions/datadog/package.json [hunk 2]
```diff
@@ -17,6 +18,29 @@
...
+      "subtitle": "Datadog",
+      "description": "List notebooks",
+      "mode": "view"
+    },
+    {
+      "name": "command-list-monitors",
+      "title": "List monitors",
```
Comment (pernielsentikaer) [inline extensions/datadog/package.json:30]:
```suggestion
      "title": "List Monitors",
```

**Review by pernielsentikaer — CHANGES_REQUESTED** [review]
Hi @aexvir 👋 

Now that @kudrykv accepted, we only some naming convention left, and then I think we're ready to proceed to final stage 🚀

Request a new review when you are ready, feel free to contact me here or at [Slack](https://raycastcommunity.slack.com/) if you have any questions.

**Comment by aexvir** [comment]
addressed and re-requested review

**Comment by pernielsentikaer** [comment]
For me it looks awesome 🔥 , can you give it a last spin @kudrykv (Since a new preference is added) - Could you help me verify that it works as expected?

**Comment by aexvir** [comment]
@kudrykv yep, more or less

our enteprise plan has a custom subdomain domain where our engineers access datadog's ui, you can read more about it on [datadog's documentation](https://docs.datadoghq.com/account_management/multi_organization/), and indeed, the api servers are the same ones as for everyone else

so while I can fetch data from our organization via api, when opening the links in the background I need to add the custom domain, otherwise I won't be able to log in and view the resources straight away

**Comment by kudrykv** [comment]
Ah, cool! Thank you for explaining it!

**Review by kudrykv — APPROVED** [review]

**Review by pernielsentikaer — APPROVED** [review]
Hi @aexvir 👋 

I'm also going to approve it, let's await final review 💪

**Review by mathieudutour — APPROVED** [review]
👏

**Pull request merged**